src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/TaskContinuation.cs

```diff
@@ -311,7 +311,7 @@ internal override void Run(Task completedTask, bool canInlineContinuationTask)
                 // If the task was cancel before running (e.g a ContinueWhenAll with a cancelled caancelation token)
                 // we will still flow it to ScheduleAndStart() were it will check the status before running
                 // We check here to avoid faulty logs that contain a join event to an operation that was already set as completed.
-                if (!continuationTask.IsCanceled && TplEventSource.Log.IsEnabled())
+                if (TplEventSource.Log.IsEnabled() && !continuationTask.IsCanceled)
```

Comment:
Does the linker not handle `something && false` should just be `false`?

Reply:
It does, it just didn't look right to me to do the call and then scrap it.

Reply:
I don’t think I understand, to do what call? This was checking a property.

Reply:
Yep but why to check the property if it's needed only when tracing is enabled. I don't feel strongly about this and could revert to the original logic event though `continuationTask.IsCanceled` is probably most of time `false`